src/components/scheduling/goals/SchedulingGoals.svelte

```diff
@@ -151,6 +151,14 @@
   }
 
   function hasDeletePermission(goal: SchedulingGoal) {
+    // If there is no spec id attached to the goal, the goal is not associated with a plan
+    // and should not be editable/deletable
+    if (
+      !goal.scheduling_specification_goal ||
+      typeof goal.scheduling_specification_goal.specification_id !== 'number'
```

Comment:
`scheduling_specification_goal` isn't typed as potentially not existing and `specification_id` is typed as only being a number. Should the type be updated to reflect these conditionals or is this a weird case?

Reply:
Yeah perhaps updating the type would be best. It is a weird case though would in theory (at least at the moment) happen any time you have goals associated with a plan and then the plan is deleted.

Reply:
Hmm...yeah if that's the case, then maybe making it clear via the types somehow would be good. When is the `specification_id` not a number?

Reply:
Or maybe another type for this specific case? I'm not sure